File: dashboard/src/main/java/de/dstoll/dashboard/service/MeetupEventProducer.java
package de.dstoll.dashboard.service;

import de.dstoll.dashboard.dto.KafkaMessage;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.kafka.core.KafkaTemplate;
import org.springframework.messaging.simp.SimpMessagingTemplate;
import org.springframework.stereotype.Service;
import org.springframework.web.util.HtmlUtils;

import java.sql.Timestamp;
import java.time.Instant;

@Service
@Slf4j
@RequiredArgsConstructor
class MeetupEventProducer {

    private final KafkaTemplate<String, String> kafkaTemplate;
    private final SimpMessagingTemplate simpMessagingTemplate;

    public void sendEvent(String event) {
        log.info("Sending message: $event");
        var timestamp = Timestamp.from(Instant.now());
        kafkaTemplate.send("meetups", event);
        simpMessagingTemplate.convertAndSend("/topic/messages",
                new KafkaMessage(timestamp, HtmlUtils.htmlEscape(event)));
    }

}

File: kafka-producer/src/main/kotlin/de/dstoll/kafkaproducer/service/MeetupEventProducer.kt
package de.dstoll.kafkaproducer.service

import de.dstoll.kafkaproducer.dto.KafkaMessage
import org.slf4j.LoggerFactory
import org.springframework.kafka.core.KafkaTemplate
import org.springframework.messaging.simp.SimpMessagingTemplate
import org.springframework.stereotype.Service
import org.springframework.web.util.HtmlUtils
import java.sql.Timestamp
import java.time.Instant

@Service
class MeetupEventProducer(
        private val kafkaTemplate: KafkaTemplate<String, String>,
        private val simpMessagingTemplate: SimpMessagingTemplate
) {

    private val log = LoggerFactory.getLogger(javaClass)

    fun sendEvent(event: String) {
        log.info("Sending message: $event")
        val timestamp = Timestamp.from(Instant.now())
        kafkaTemplate.send("meetups", event)
        simpMessagingTemplate.convertAndSend("/topic/messages",
                KafkaMessage(timestamp, HtmlUtils.htmlEscape(event)))
    }

}